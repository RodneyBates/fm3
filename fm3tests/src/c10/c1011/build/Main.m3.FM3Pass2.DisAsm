 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L1(16_1),[11,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenScopeLt(L1(16_1))
 ItkOpenDeclListLt(L6(16_6),[13,2])
 ItkConstDeclLt([13,8])
 ItkDeclNo(D1,[13,8])
 ItkConstDeclType([13,10])
 ItkDeclTypeAbsent([13,10])
 ItkConstDeclValue([13,10])
 ItkDefTopExprNo(L8(16_8),[13,12])
 ItkConstDeclRt([13,8])
 ItkOpenDeclListSep(L1(16_1),[13,2])
 ItkOpenDeclListSep(L1(16_1),[13,2])
 ItkConstDeclLt([15,8])
 ItkDeclNo(D2,[15,8])
 ItkConstDeclType([15,10])
 ItkDefTopExprNo(L7(16_7),[15,12])
 ItkConstDeclValue([15,20])
 ItkDefTopExprNo(L6(16_6),[15,22])
 ItkConstDeclRt([15,8])
 ItkOpenDeclListSep(L2(16_2),[13,2])
 ItkOpenDeclListSep(L2(16_2),[13,2])
 ItkConstDeclLt([17,8])
 ItkDeclNo(D3,[17,8])
 ItkConstDeclType([17,10])
 ItkDeclTypeAbsent([17,10])
 ItkConstDeclValue([17,10])
 ItkDefTopExprNo(L5(16_5),[17,12])
 ItkConstDeclRt([17,8])
 ItkOpenDeclListSep(L3(16_3),[13,2])
 ItkConstDeclLt([20,4])
 ItkDeclNo(D4,[20,4])
 ItkConstDeclType([20,7])
 ItkDeclTypeAbsent([20,7])
 ItkConstDeclValue([20,7])
 ItkDefTopExprNo(L4(16_4),[20,9])
 ItkConstDeclRt([20,4])
 ItkOpenDeclListSep(L1(16_1),[21,2])
 ItkConstDeclLt([22,4])
 ItkDeclNo(D5,[22,4])
 ItkConstDeclType([22,7])
 ItkDefTopExprNo(L3(16_3),[22,9])
 ItkConstDeclValue([22,14])
 ItkDefTopExprNo(L2(16_2),[22,16])
 ItkConstDeclRt([22,4])
 ItkOpenDeclListSep(L2(16_2),[23,2])
 ItkConstDeclLt([24,4])
 ItkDeclNo(D6,[24,4])
 ItkConstDeclType([24,7])
 ItkDeclTypeAbsent([24,7])
 ItkConstDeclValue([24,7])
 ItkDefTopExprNo(L1(16_1),[24,9])
 ItkConstDeclRt([24,4])
 ItkOpenDeclListRt(L6(16_6),[13,2])
 ItkModuleBegin(L1(16_1),[26,2])
 ItkStmtListLt(L0(16_0),[27,2])
 ItkStmtListRt(L0(16_0),[27,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L1(16_1),[11,0])
 ItkRightEnd()
 ItkBOF()
     0
