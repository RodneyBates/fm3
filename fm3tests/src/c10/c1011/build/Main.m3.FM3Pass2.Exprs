From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [24,9]
 ExpOpcode = FALSE
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 2 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 1067030938
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [22,16]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 3 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 3
 ExpPosition = [22,9]
 ExpOpcode = REAL
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 4 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 11
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 4
 ExpPosition = [20,9]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 5 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 5
 ExpPosition = [17,12]
 ExpOpcode = TRUE
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 6 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 15
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 6
 ExpPosition = [15,22]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 7 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 7
 ExpPosition = [15,12]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


