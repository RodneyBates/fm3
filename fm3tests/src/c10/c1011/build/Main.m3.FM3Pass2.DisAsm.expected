 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L2(16_2),[11,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenScopeLt(L1(16_1))
 ItkOpenDeclListLt(L5(16_5),[13,2])
 ItkConstDeclLt([15,8])
 ItkDeclNo(D3,[15,8])
 ItkConstDeclType([15,10])
 ItkDefTopExprNo(L7(16_7),[15,12])
 ItkConstDeclValue([15,20])
 ItkDefTopExprNo(L6(16_6),[15,22])
 ItkConstDeclRt([15,8])
 ItkOpenDeclListSep(L1(16_1),[13,2])
 ItkOpenDeclListSep(L1(16_1),[13,2])
 ItkConstDeclLt([17,8])
 ItkDeclNo(D4,[17,8])
 ItkConstDeclType([17,10])
 ItkDeclTypeAbsent([17,10])
 ItkConstDeclValue([17,10])
 ItkDefTopExprNo(L5(16_5),[17,12])
 ItkConstDeclRt([17,8])
 ItkOpenDeclListSep(L2(16_2),[13,2])
 ItkConstDeclLt([20,4])
 ItkDeclNo(D5,[20,4])
 ItkConstDeclType([20,7])
 ItkDeclTypeAbsent([20,7])
 ItkConstDeclValue([20,7])
 ItkDefTopExprNo(L4(16_4),[20,9])
 ItkConstDeclRt([20,4])
 ItkOpenDeclListSep(L1(16_1),[21,2])
 ItkConstDeclLt([22,4])
 ItkDeclNo(D6,[22,4])
 ItkConstDeclType([22,7])
 ItkDefTopExprNo(L3(16_3),[22,9])
 ItkConstDeclValue([22,14])
 ItkDefTopExprNo(L2(16_2),[22,16])
 ItkConstDeclRt([22,4])
 ItkOpenDeclListSep(L2(16_2),[23,2])
 ItkConstDeclLt([24,4])
 ItkDeclNo(D7,[24,4])
 ItkConstDeclType([24,7])
 ItkDeclTypeAbsent([24,7])
 ItkConstDeclValue([24,7])
 ItkDefTopExprNo(L1(16_1),[24,9])
 ItkConstDeclRt([24,4])
 ItkOpenDeclListRt(L5(16_5),[13,2])
 ItkModuleBegin(L2(16_2),[26,2])
 ItkStmtListLt(L0(16_0),[27,2])
 ItkStmtListRt(L0(16_0),[27,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L2(16_2),[11,0])
 ItkRightEnd()
 ItkBOF()
     0
